[Offline sync] New features not synced to remote server

Stack trace:
```
2019-09-11 21:19:39.766 10558-10702/com.google.android.gnd D/LocalMutationSyncWorker: Remote updates for feature ThmJBl6ksh5yPz9oBrB8 failed
    java.lang.IllegalStateException: Cannot access database on the main thread since it may potentially lock the UI for a long period of time.
        at androidx.room.RoomDatabase.assertNotMainThread(RoomDatabase.java:261)
        at androidx.room.RoomDatabase.compileStatement(RoomDatabase.java:315)
        at com.google.android.gnd.persistence.local.room.FeatureMutationDao_Impl$6.call(FeatureMutationDao_Impl.java:258)
        at com.google.android.gnd.persistence.local.room.FeatureMutationDao_Impl$6.call(FeatureMutationDao_Impl.java:249)
        at io.reactivex.internal.operators.completable.CompletableFromCallable.subscribeActual(CompletableFromCallable.java:36)
        at io.reactivex.Completable.subscribe(Completable.java:2309)
        at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly.subscribeActual(CompletableOnAssembly.java:39)
        at io.reactivex.Completable.subscribe(Completable.java:2309)
        at io.reactivex.internal.operators.completable.CompletableAndThenCompletable.subscribeActual(CompletableAndThenCompletable.java:35)
        at io.reactivex.Completable.subscribe(Completable.java:2309)
        at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly.subscribeActual(CompletableOnAssembly.java:39)
        at io.reactivex.Completable.subscribe(Completable.java:2309)
        at io.reactivex.internal.operators.completable.CompletableAndThenCompletable$SourceObserver.onComplete(CompletableAndThenCompletable.java:67)
        at com.akaita.java.rxjava2debug.extensions.CompletableOnAssembly$OnAssemblyCompletableObserver.onComplete(CompletableOnAssembly.java:71)
        at io.reactivex.internal.operators.completable.CompletableCreate$Emitter.onComplete(CompletableCreate.java:64)
        at com.google.android.gnd.rx.RxTask.lambda$null$4(RxTask.java:74)
        at com.google.android.gnd.rx.-$$Lambda$RxTask$8seOJ-bOiPEAknEl_9ZxHw1hfSc.onSuccess(Unknown Source:2)
        at com.google.android.gms.tasks.zzn.run(Unknown Source:27)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
     Caused by: com.akaita.java.rxjava2debug.extensions.RxJavaAssemblyException
     Caused by: com.akaita.java.rxjava2debug.extensions.RxJavaAssemblyException
     Caused by: com.akaita.java.rxjava2debug.extensions.RxJavaAssemblyException
```

--- a/gnd/src/main/java/com/google/android/gnd/persistence/sync/LocalMutationSyncWorker.java
+++ b/gnd/src/main/java/com/google/android/gnd/persistence/sync/LocalMutationSyncWorker.java
@@ -30,6 +30,7 @@ import com.google.android.gnd.persistence.local.LocalDataStore;
 import com.google.android.gnd.persistence.remote.RemoteDataStore;
 import com.google.common.collect.ImmutableList;
 import io.reactivex.Completable;
+import io.reactivex.schedulers.Schedulers;
 
 /**
  * A worker that syncs local changes to the remote data store. Each instance handles mutations for a
@@ -77,9 +78,11 @@ public class LocalMutationSyncWorker extends Worker {
   }
 
   private Completable processMutations(ImmutableList<Mutation> pendingMutations) {
+    // TODO(#178): Why is removePendingMutations() getting run on UI thread without subscribeOn()?
     return remoteDataStore
         .applyMutations(pendingMutations)
-        .andThen(localDataStore.removePendingMutations(pendingMutations));
+        .andThen(
+            localDataStore.removePendingMutations(pendingMutations).subscribeOn(Schedulers.io()));
   }
 
   private ImmutableList<Mutation> incrementRetryCounts(

